src/java.base/unix/classes/sun/nio/fs/UnixFileSystem.java

```diff
@@ -1027,7 +1029,10 @@ void copy(final UnixPath source,
             try {
                 // the access(2) system call always follows links so it
                 // is suppressed if the source is an unfollowed link
-                access(source, R_OK);
+                int errno = access(source, R_OK);
+                if (errno != 0)
+                    throw new UnixException(errno);
```

Comment:
Are you going to change this one to use `new UnixException(errno).rethrowAsIOException(source);` too?

Reply:
Yes. It was overlooked. Thanks.